java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -690,6 +690,215 @@ void testInnerJoinWithOnlyCommonKeys() {
+
+    boundsUnequalCols2(false);
+
+    boundsNulls(false);
+  }
+
+  @Test
+  void testUpperBound() {
+    boundsEmptyInput(true);
+
+    boundsEmptyValues(true);
+
+    boundsStringCategories(true);
+
+    boundsIntsSingleCol(true);
+
+    boundsLongsSingleCol(true);
+
+    boundsLongsMultiVal(true);
+
+    boundsMultiCol(true);
+
+    boundsUnequalCols1(true);
+
+    boundsUnequalCols2(true);
+
+    boundsNulls(true);
+  }
+
+  private void boundsNulls(boolean isUpperBound) {
+    boolean[] descFlags = new boolean[3];
+    try(Table table = new TestBuilder()
+            .column(null, 20, 20, 20, 30)
+            .build();
+        Table values = new TestBuilder()
+            .column(15)
+            .build();
+        ColumnVector columnVector = getBoundsCv(descFlags, isUpperBound, table, values);
+        ColumnVector expected = ColumnVector.fromBoxedInts(1)) {
+      assertColumnsAreEqual(expected, columnVector);
+    }
+  }
+
+  private void boundsUnequalCols2(boolean isUpperBound) {
+    boolean[] descFlags = new boolean[3];
+    try(Table table = new TestBuilder()
+            .column(90, 100, 120, 130, 135)
+            .column(.5, .5, .5, .7, .7)
+            .build();
+        Table values = new TestBuilder()
+            .column(120)
+            .column(.3)
+            .column(.7)
+            .build();
+        ColumnVector columnVector = getBoundsCv(descFlags, isUpperBound, table, values);
+        ColumnVector expected = ColumnVector.fromBoxedInts(2)) {
+      assertColumnsAreEqual(expected, columnVector);
+    }
+  }
+
+  private void boundsUnequalCols1(boolean isUpperBound) {
+    boolean[] descFlags = new boolean[3];
+    try(Table table = new TestBuilder()
+            .column(90, 100, 120, 130, 135)
+            .column(.5, .5, .5, .7, .7)
+            .column(90, 100, 120, 130, 135)
+            .build();
+        Table values = new TestBuilder()
+            .column(120)
+            .column(.3)
+            .build();
+        ColumnVector columnVector = getBoundsCv(descFlags, isUpperBound, table, values);
+        ColumnVector expected = ColumnVector.fromBoxedInts(2)) {
+      assertColumnsAreEqual(expected, columnVector);
+    }
+  }
+
+  private void boundsMultiCol(boolean isUpperBound) {
+    boolean[] descFlags = new boolean[3];
+    try(Table table = new TestBuilder()
+            .column(10, 20, 20, 20, 20)
+            .column(5.0, .5, .5, .7, .7)
+            .column(90, 77, 78, 61, 61)
+            .build();
+        Table values = new TestBuilder()
+            .column(20)
+            .column(0.7)
+            .column(61)
+            .build();
+        ColumnVector columnVector = getBoundsCv(descFlags, isUpperBound, table, values);
+        ColumnVector expected = isUpperBound ? ColumnVector.fromBoxedInts(5) :
+            ColumnVector.fromBoxedInts(3)) {
+      assertColumnsAreEqual(expected, columnVector);
+    }
+  }
+
+  private void boundsLongsMultiVal(boolean isUpperBound) {
```

Comment:
This one and the next one say they are for longs, but they have floating values. Should it be `boundsFloatMultiVal`?